### ISSUE TITLE
legion-20.03.0: scatter test fails on armv7hl

### ISSUE THREAD

**Issue by junghans** [StanfordLegion/legion#803]
From https://koji.fedoraproject.org/koji/taskinfo?taskID=42940047:
```
      Start 36: scatter
36/41 Test #36: scatter ..........................Child aborted***Exception:   3.21 sec
scatter: /builddir/build/BUILD/legion-legion-20.03.0/runtime/realm/transfer/channel.cc:1509: void Realm::XferDes::update_pre_bytes_write(int, size_t, size_t, size_t): Assertion `val == pre_bytes_total' failed.
```

**Comment by streichler** [comment]
@junghans did this work on legion-19.12.0?  I don't have access to an arm test setup right now.  Do you have a way to reproduce this on a local system?

**Comment by junghans** [comment]
It passed with 19.12.0.

**Comment by streichler** [comment]
It turns out I can reproduce this error in i686 builds as well - debugging now.

**Comment by streichler** [comment]
@junghans can you see if this patch fixes the problem:
https://gitlab.com/StanfordLegion/legion/-/commit/0741659dc1961b2cdfc7a6c33015670eec2942f3.diff

**Comment by junghans** [comment]
Running: https://koji.fedoraproject.org/koji/taskinfo?taskID=42951007

**Comment by junghans** [comment]
@streichler  yes that fixes armv7hl, i686 is still broken though.

**Comment by streichler** [comment]
Fix is in the master branch with commit 7db9c79b.
